tests/foreman/ui/test_reporttemplates.py

```diff
@@ -396,23 +138,91 @@ def test_positive_e2e_crud(session, module_org, module_loc):
             'organizations.resources.assigned': [module_org.name],
             'locations.resources.assigned': [module_loc.name],
         })
-        assert session.reporttemplate.search(name)[0]['Name'] == name
-        pt = session.reporttemplate.read(name)
-        assert pt['template']['name'] == name
-        assert pt['template']['default'] is True
-        assert pt['template']['template_editor']['editor'] == content
-        assert pt['inputs'][0]['name'] == input_name
-        assert pt['inputs'][0]['required'] is True
-        assert pt['inputs'][0]['input_type'] == 'User input'
-        assert pt['type']['snippet'] is False
-        assert pt['locations']['resources']['assigned'][0] == module_loc.name
-        assert pt['organizations']['resources']['assigned'][0] == module_org.name
+        # READ report template
+        rt = session.reporttemplate.read(name)
+        assert rt['template']['name'] == name
+        assert rt['template']['default'] is True
+        assert rt['template']['template_editor']['editor'] == content
+        assert rt['inputs'][0]['name'] == input_name
+        assert rt['inputs'][0]['required'] is True
+        assert rt['inputs'][0]['input_type'] == 'User input'
+        assert rt['type']['snippet'] is False
+        assert rt['locations']['resources']['assigned'][0] == module_loc.name
+        assert rt['organizations']['resources']['assigned'][0] == module_org.name
+        # UPDATE report template
         session.reporttemplate.update(
             name,
             {'template.name': new_name, 'type.snippet': True}
         )
         assert session.reporttemplate.search(new_name)[0]['Name'] == new_name
-        updated_pt = session.reporttemplate.read(new_name)
-        assert updated_pt['type']['snippet'] is True
+        rt = session.reporttemplate.read(new_name)
+        assert rt['type']['snippet'] is True
+        # LOCK
+        session.reporttemplate.lock(new_name)
+        assert session.reporttemplate.is_locked(new_name) is True
+        # UNLOCK
+        session.reporttemplate.unlock(new_name)
+        assert session.reporttemplate.is_locked(new_name) is False
+        # CLONE
+        session.reporttemplate.clone(
+            new_name,
+            {
+                'template.name': clone_name,
+            }
+        )
+        rt = entities.ReportTemplate().search(
```

Comment:
same, why do you use API there?

Reply:
I think the main disagreement here is @lhellebr says entity's **read** via UI is tested because it is done in `# READ report template` section and do not need to be tested again. On the other hand, @oshtaier says that **every entity** change done on in UI test needs to be done via UI - e.g. cloned entity might be unreadable via UI. Another Lukas's argument is he wants to verify change via another means that what caused the change to make sure (when thinking about it "ad absurdum"), UI might clone the entity only in data backend used by UI and not by API.
In ideal case, we would be checking with both UI and API, but that would be very expensive, so reasonable mix of these two ways seems to be a best solution to me.

Reply:
@jhutar 
> In ideal case, we would be checking with both UI and API
We do not even both ways, but all three (including CLI), so we test *every entity* by *API, CLI, UI*, so I have no idea why we mix ways when we still validate same thing in API